docs/03-language-guide/14-equality.md

```diff
@@ -1,6 +1,133 @@
 ---
 title: Equality
 id: equality
+description: A reference guide to equality in Wing.
+keywords: [equality, equals, identity]
 ---
 
-TODO (for now, see the [language spec RFC](../999-contributors/999-rfcs/2023-06-12-language-spec.md))
+Checking for equality is performed with the `==` operator. It returns `true` if the two values are equal, and `false` otherwise.
+
+The main difference between equality in JavaScript and Wing is that `==` in Wing is not allowed to compare values of different types. For example, `1 == "1"` is not allowed in Wing, and will result in a compile-time error.
+
+The execution phase ([preflight or inflight](../02-core-concepts/01-preflight-and-inflight.md)) that a value was created in does not affect its equality. For example, a value created in preflight can be equal to a value created in inflight.
+
+Equality in Wing is a symmetric and transitive relationship - that is, (1) if `a == b`, then `b == a`, and (2) if `a == b` and `b == c`, then `a == c`.
+
+The following is a set of rules for checking equality:
+
```

Comment:
I think it's worth defining the terms "by value" and "by reference" here and using the at the beginning of each section.

Also worth mentioning the concept of deep versus shallow comparison.

Reply:
Good idea, I think adding "by value" / "by reference" improves the explanation 👍

the way I've defined the semantics hopefully doesn't require burdening users with understanding the difference between deep and shallow comparisons (the suggested direction is if we're checking if two arrays are equal, we perform `==` on every element recursively); but if that doesn't end up working, then we can add it back.